packages/exchange-core/src/test/Matching.test.ts

```diff
@@ -102,21 +105,38 @@ describe('Matching tests', () => {
     const cloneOrder = (order: Order, traded: number) => {
```

Comment:
This should be a generic method, where `T extends Order`. Like:

```
function cloneOrder<T extends Order>(order: T, trader: number)
```

Then you don't need to cast the return of it to either `Bid` or `Ask`, as is currently seen in the code. Some examples from current codebase:

```
const bidsAfter = [cloneOrder(bidsBefore[1], onekWh) as Bid];
// ...
const asksAfter = [cloneOrder(asksBefore[0], onekWh * 3) as Ask, asksBefore[1]];
```

Example implementation of generic method:
```
interface Order {
    order: string;
}

interface Bid extends Order {
    bid: string;
}

function cloneOrder<T extends Order>(order: T, trader: number): T {
    return order;
}

const bid: Bid = null;

const result = cloneOrder(bid, 1); // result will be of type Bid
```

Reply:
This won't really work in this case because you need to create new instance of the object (using non-default constructor). The way to go is actually to provide clone() function on Bid and Ask class.

Reply:
I'm not talking about changing any of the method's logic. Only types. I actually provided signature which should be a drop-in replacement:

```
cloneOrder<T extends Order>(order: T, trader: number): T
```

It's worth to try testing it. No need to change any of the method's logic.